src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Logger, SfdxError, LoggerLevel} from '@salesforce/core';
+import {Catalog, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, ESRule, ESReport, ESMessage} from '../../types';
+import {OutputProcessor} from '../pmd/OutputProcessor';
+import {RuleEngine} from '../services/RuleEngine';
+import {CLIEngine} from 'eslint';
+import * as path from 'path';
+
+
+export interface EslintStrategy {
+
+	init(): Promise<void>;
+
+	getName(): string;
+
+	isEnabled(): boolean;
+
+	getTargetPatterns(target?: string): Promise<string[]>;
+
+	getCatalogConfig(): Record<string, any>;
+
+	getRunConfig(target?: string): Promise<Record<string, any>>;
+
+	getLanguage(): string[];
+
+	isRuleKeySupported(key: string): boolean;
+
+	filterUnsupportedPaths(paths: string[]): string[];
+}
+export class StaticDependencies {
+	public createCLIEngine(config: Record<string,any>): CLIEngine {
+		return new CLIEngine(config);
+	}
+	
+	public resolveTargetPath(target: string): string {
+		return path.resolve(target);
+	}
+	
+	public getCurrentWorkingDirectory(): string {
+		return process.cwd();
+	}
+}
+
+export abstract class BaseEslintEngine implements RuleEngine {
```

Comment:
This class was originally ESLintEngine that handled both Javascript and Typescript. It would make decisions on which logic to use based on presence or absence of tsconfig.json file. This caused a few different if/else based issues that were getting difficult to track.
Here, I've put all the common logic into the abstract BaseEslintEngine class. All the decision points are outsourced to EslintStrategy interface (defined above). The decision points are implemented differently for JavascriptEslintStrategy and TypescriptEslintStrategy.